src/ui/MissingTerm/MissingTerm.ts

```diff
@@ -0,0 +1,102 @@
+import 'styling/_MissingTerm';
+import { exportGlobally } from '../../GlobalExports';
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { reject } from 'underscore';
+import { $$, Initialization } from '../../Core';
+import { Dom } from '../../utils/Dom';
+import { analyticsActionCauseList, IAnalyticsIncludeMissingTerm } from '../Analytics/AnalyticsActionListMeta';
+
+export interface IMissingTermOptions {
+  clickable?: boolean;
+}
+
+export class MissingTerm extends Component {
+  static ID = 'MissingTerm';
+  static options: IMissingTermOptions = {
+    /**
+     * Enable the user to click on a missing term to force it in the query
+     */
+    clickable: ComponentOptions.buildBooleanOption({ defaultValue: true })
+  };
+
+  static doExport = () => {
+    exportGlobally({
+      MissingTerm: MissingTerm
+    });
+  };
+  /**
+   * The `MissingTerm` component is responsible for displaying the keyword that was not matched for a result
+   */
+  constructor(public element: HTMLElement, public options?: IMissingTermOptions, bindings?: IComponentBindings, public result?) {
+    super(element, MissingTerm.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerm, options);
+    this.addMissingTerm();
+  }
+
+  private addMissingTerm() {
+    if (this.getMissingTerms().length > 0) {
+      const container = this.buildContainer();
+      $$(this.element).append(container.el);
+      const cell = $$(this.element).find('.coveo-result-cell');
```

Comment:
It's weird to build the container, then query back on it to do further modification. Not as straightforward as it could be.

Would be much better if the code was arranged in a way where you build the `coveo-result-cell` properly inside `buildContainer()`, with the caption already included.

Reply:
You're right here's what I'll do
```
private addMissingTerm() {
    if (this.missingTerms.length === 0) {
      return ;
    }
    $$(this.element).append(this.buildContainer().el);
  }

private buildContainer(): Dom {
    const resultCell = $$('div', { className: 'coveo-result-cell' }, this.buildCaption())
    this.buildMissingTerms().forEach(term => {
      resultCell.append(term.el);
    });

    return $$('div', { className: 'coveo-result-row' }, resultCell);
  };
```
Any Comments?

Reply:
Looks good, 💯
Thanks